### PR TITLE
Feature/bulk operations

### PR DESCRIPTION
**Pull request by nk-coding**

### CODE REVIEW

File: core/src/main/kotlin/gropius/service/architecture/ComponentService.kt [hunk 1]
```diff
@@ -61,6 +63,37 @@ class ComponentService(
...
+
+    /**
+     * Creates a new [Component] based on the provided [input]
+     * Does not check the authorization status
+     *
+     * @param input defines the [Component]
+     * @param user the user who created the [Component]
+     * @return the saved created [Component]
+     */
+    private suspend fun createComponentInternal(input: CreateComponentInput, user: GropiusUser, ): Component {
```
Comment (chriku) [inline core/src/main/kotlin/gropius/service/architecture/ComponentService.kt:96]:
```suggestion
    private suspend fun createComponentInternal(input: CreateComponentInput, user: GropiusUser): Component {
```

File: core/src/main/kotlin/gropius/service/architecture/RelationService.kt [hunk 2]
```diff
@@ -75,6 +76,22 @@ class RelationService(
...
+     * @param authorizationContext used to check for the required permission
+     * @param input defines the [Relation]s
+     * @return the saved created [Relation]s
+     */
+    suspend fun bulkCreateRelation(
+        authorizationContext: GropiusAuthorizationContext, input: BulkCreateRelationInput
+    ): List<Relation> {
+        input.validate()
+        val relations = input.relations.map { createRelation(authorizationContext, it) }
+        return relations
```
Comment (chriku) [inline core/src/main/kotlin/gropius/service/architecture/RelationService.kt:92]: aren't most of the relations ded if you din't hand through the dfe?

**Review by chriku — CHANGES_REQUESTED** [review]

File: core/src/main/kotlin/gropius/service/architecture/RelationService.kt (hunk 2 above)
Comment (nk-coding) [reply core/src/main/kotlin/gropius/service/architecture/RelationService.kt:92]: I don't really understand this comment. All mutations use the dfe just for authorization information

Comment (chriku) [reply core/src/main/kotlin/gropius/service/architecture/RelationService.kt:92]: brainfart. meant nodeCache and that's irrelevant, because nit saved

**Review by chriku — APPROVED** [review]

**Pull request merged**